File: src/test/java/com/zepben/cimbend/database/sqlite/GenerateSqliteSql.java
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.cimbend.database.sqlite;

import com.zepben.cimbend.database.sqlite.tables.SqliteTable;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static com.zepben.cimbend.common.interop.JavaLambda.unit;

public class GenerateSqliteSql {

    private static final Logger logger = LoggerFactory.getLogger(GenerateSqliteSql.class);

    @Test
    @Disabled
    public void runThisFunctionAsATestToGetSqlForAnSqliteDatabase() {
        new DatabaseTables().forEachTable(unit(this::printSqlFields));
    }

    private void printSqlFields(SqliteTable table) {
        logger.info(table.createTableSql() + ";");
        logger.info("");

        for (String sql : table.createIndexesSql())
            logger.info(sql + ";");
        logger.info("");

        logger.info(table.preparedInsertSql() + ";");
        logger.info("");

        logger.info(table.selectSql() + ";");
        logger.info("");

        logger.info(table.preparedUpdateSql() + ";");
        logger.info("");
    }

}

File: src/test/kotlin/com/zepben/evolve/database/sqlite/GenerateSqliteSql.kt
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.zepben.evolve.database.sqlite

import com.zepben.evolve.database.sqlite.cim.customer.CustomerDatabaseTables
import com.zepben.evolve.database.sqlite.cim.diagram.DiagramDatabaseTables
import com.zepben.evolve.database.sqlite.cim.network.NetworkDatabaseTables
import com.zepben.evolve.database.sqlite.cim.tables.SqliteTable
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

class GenerateSqliteSql {

    //
    // NOTE: These tests are deliberately disabled as they simply print the database SQL statements which
    //       just clogs up the test log. You should selectively run them if you need the statements.
    //

    @Test
    @Disabled
    fun `print database create statements`() {
        printStatements { table ->
            println(table.createTableSql + ";")
            table.createIndexesSql.forEach { println("$it;") }
            println("")
        }
    }

    @Test
    @Disabled
    fun `print database select statements`() {
        printStatements { table ->
            println(table.selectSql + ";")
        }
    }

    @Test
    @Disabled
    fun `print database insert statements`() {
        printStatements { table ->
            println(table.preparedInsertSql + ";")
        }
    }

    @Test
    @Disabled
    fun `print database update statements`() {
        printStatements { table ->
            println(table.preparedUpdateSql + ";")
        }
    }

    private fun printStatements(action: (SqliteTable) -> Unit) {
        println("******** Customer Database ********")
        println("")
        CustomerDatabaseTables().forEachTable(action)
        println("")
        println("******** Diagram Database ********")
        println("")
        DiagramDatabaseTables().forEachTable(action)
        println("")
        println("******** Network Database ********")
        println("")
        NetworkDatabaseTables().forEachTable(action)
        println("")
    }

}
